lisa/operating_system.py

```diff
@@ -948,37 +948,31 @@ class OpenBSD(BSD):
 
 
 @dataclass
-# yum repolist is of the form `<id> <name>`
+# dnf repolist is of the form `<id> <name>`
 # Example:
 # microsoft-azure-rhel8-eus  Microsoft Azure RPMs for RHEL8 Extended Update Support
-class FedoraRepositoryInfo(RepositoryInfo):
+class RPMRepositoryInfo(RepositoryInfo):
     # id for the repository, for example: microsoft-azure-rhel8-eus
     id: str
 
 
-class Fedora(Linux):
-    # Red Hat Enterprise Linux Server 7.8 (Maipo) => 7.8
-    _fedora_release_pattern_version = re.compile(r"^.*release\s+([0-9\.]+).*$")
-
+# Linux distros that use RPM.
+class RPMDistro(Linux):
     # microsoft-azure-rhel8-eus  Microsoft Azure RPMs for RHEL8 Extended Update Support
-    _fedora_repository_info_pattern = re.compile(r"(?P<id>\S+)\s+(?P<name>\S.*\S)\s*")
+    _rpm_repository_info_pattern = re.compile(r"(?P<id>\S+)\s+(?P<name>\S.*\S)\s*")
 
     # ex: dpdk-20.11-3.el8.x86_64
-    _fedora_version_splitter_regex = re.compile(
+    _rpm_version_splitter_regex = re.compile(
         r"(?P<package_name>[a-zA-Z0-9\-_]+)-"
         r"(?P<major>[0-9]+)\."
         r"(?P<minor>[0-9]+)"
         r"(?P<patch>\.[0-9]+)?"
         r"(?P<build>-[a-zA-Z0-9-_\.]+)?"
     )
 
-    @classmethod
-    def name_pattern(cls) -> Pattern[str]:
-        return re.compile("^Fedora|fedora$")
-
     def get_repositories(self) -> List[RepositoryInfo]:
         repo_list_str = self._node.execute(
-            "yum repolist", sudo=True
+            "dnf repolist", sudo=True
```

Comment:
dnf is not supported by some Redhat 7.x versions. So, it's using the yum. Can you extract the tool name as a variable, use "yum" by default, and overwrites it to "dnf" in Mariner?

Reply:
Ideally, the package management should be split from the os, but it's not done. We can take this as workaround.

Reply:
Why does `_install_packages` use `dnf` if it isn't supported?

Reply:
You are right. Some methods are overridden in RedHat. It's not a thoughtful solution, and should be refined later. For now, the cheapest solution is to extract the "dnf" or "yum" out to me, if used commands are compatible.